lib/inferno/repositories/results.rb

```diff
@@ -85,6 +166,19 @@ def validate
           super
           errors.add(:result, "'#{result}' is not valid") unless Entities::Result::RESULT_OPTIONS.include?(result)
         end
+
+        def self.current_results_for_test_session(test_session_id)
+          fetch(CURRENT_RESULTS_SQL, test_session_id)
+        end
+
+        def self.current_child_results_for_test_session(test_session_id, test_ids, test_group_ids)
```

Comment:
I'm a little confused what this does.  Is it just that it lets you filter to a subset of test_ids and group_ids on the server side?  I would think you'd just use the "current_results_for_test_session" method and just have the group/test ids be optional.  It might be a tad tricky to formulate a frozen string for the sql command that is smart enough to know that the array passed to it is null -- depends on the SQL gem.  If we think this query is complex, it probably would be good to condense into one copy instead of two slightly different versions (its hard to me to see what is different about them?

Reply:
I could change it so that this method determines the children rather than having them passed in. As you say, the name is confusing because there's nothing child specific in this code.

We could potentially unify the queries with something like:
```
def current_results_for_test_session(test_session_id, test_ids: [], test_group_ids: [], test_suite_ids: [])
  query = fetch(CURRENT_RESULTS_SQL, test_session_id)
  return query if test_ids.blank? && group_ids.blank? && test_suite_ids.blank?

  query.where(
    Sequel[test_id: test_ids] |
       Sequel[test_group_ids: test_group_ids] |
       Sequel[test_suite_ids: test_suite_ids]
  )
end
```

Reply:
I couldn't get anything like the above to work with the Sequel DSL, but I was able to make the big queries more unified.